packages/core/src/application_ref.ts

```diff
@@ -770,66 +773,7 @@ export class ApplicationRef {
   }
 
   /** @internal */
-  constructor(
-      private _zone: NgZone,
-      private _injector: EnvironmentInjector,
-      private _exceptionHandler: ErrorHandler,
-  ) {
-    this._onMicrotaskEmptySubscription = this._zone.onMicrotaskEmpty.subscribe({
-      next: () => {
-        this._zone.run(() => {
-          this.tick();
-        });
-      }
-    });
-
-    const isCurrentlyStable = new Observable<boolean>((observer: Observer<boolean>) => {
-      this._stable = this._zone.isStable && !this._zone.hasPendingMacrotasks &&
-          !this._zone.hasPendingMicrotasks;
-      this._zone.runOutsideAngular(() => {
-        observer.next(this._stable);
-        observer.complete();
-      });
-    });
-
-    const isStable = new Observable<boolean>((observer: Observer<boolean>) => {
-      // Create the subscription to onStable outside the Angular Zone so that
-      // the callback is run outside the Angular Zone.
-      let stableSub: Subscription;
-      this._zone.runOutsideAngular(() => {
-        stableSub = this._zone.onStable.subscribe(() => {
-          NgZone.assertNotInAngularZone();
-
-          // Check whether there are no pending macro/micro tasks in the next tick
-          // to allow for NgZone to update the state.
-          scheduleMicroTask(() => {
-            if (!this._stable && !this._zone.hasPendingMacrotasks &&
-                !this._zone.hasPendingMicrotasks) {
-              this._stable = true;
-              observer.next(true);
-            }
-          });
-        });
-      });
-
-      const unstableSub: Subscription = this._zone.onUnstable.subscribe(() => {
-        NgZone.assertInAngularZone();
-        if (this._stable) {
-          this._stable = false;
-          this._zone.runOutsideAngular(() => {
-            observer.next(false);
-          });
-        }
-      });
-
-      return () => {
-        stableSub.unsubscribe();
-        unstableSub.unsubscribe();
-      };
-    });
-
-    this.isStable = merge(isCurrentlyStable, isStable.pipe(share()));
-  }
+  constructor(private _injector: EnvironmentInjector) {}
```

Comment:
May be move this outside of a constructor?
```suggestion
  private injector = inject(EnvironmentInjector);
```